src/main/clojure/clara/rules/compiler.clj

```diff
@@ -235,15 +235,21 @@
   "Compiles a sequence of constraints into a structure that can be evaluated.
 
    Callers may also pass a collection of equality-only-variables, which instructs
-   this function to only do an equality check on them rather than create a unification binding."
+   this function to only do an equality check on them rather than create a unification binding.
+
+   The returned expression evaluates to a two item vector, [bindings error], which contains
+   the evaluated bindings in the first argument and a map with information on any exception
+   that might have been thrown by evaluating a constraint as the second vector entry."
```

Comment:
Apologies for being slow here...I had been traveling and then got swamped with other items.

Overall I think this looks good. One thing that's not clear to me: rather than returning a [bindings error] tuple, could the compiled function simple return the bindings (as it did before) or throw a specific type of an exception that the caller could handle when an error occurs? The thrown exception could include the same information as seen in the error in the returned tuple, so we wouldn't lose anything.

It's probably not a huge deal, but throw/catching a specialized exception here would save us from creating and de-structuring an extra for every time we run our constraint expression.